File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int size = 0;
        ListNode temp = head;
        while(temp != null){
            temp = temp.next;
            size ++;
        }
        if(n == 1 && size ==1)
            return null;
        
        if(n == size){
            return head.next;
        }
        
        int count = 1;
        temp = head;
        int num = (size - n + 1);
        while(temp != null){
            
            if(count == num-1){
                temp.next = temp.next.next;
                break;
            }
            temp = temp.next;
            count++;
        }
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int i = 0;
        ListNode* fast = head;
        while(i != n){
            i++;
            fast = fast->next;
        }
        if(fast == nullptr)
            return head->next;
        ListNode* slow = head;
        while(fast->next){
            slow = slow->next;
            fast = fast->next;
        }
        ListNode* next = slow->next;
        slow->next = next->next;
        delete next;
        return head;
    }
};